crates/iota-genesis-builder/src/stardust/process_outputs.rs

```diff
@@ -67,9 +81,253 @@ pub fn is_participation_output(output: &Output) -> bool {
+                                // meet.
+                                let (original_output_opt, split_outputs) = swap_split_operation(
+                                    destinations.iter_by_tokens_target_mut_filtered(),
+                                    basic_output,
+                                );
+                                // if some SwapSplit were performed, their result are basic inputs
+                                // stored in split_outputs; so, we save them in
+                                // split_basic_outputs to return them later
+                                if !split_outputs.is_empty() {
+                                    self.num_swapped_basic += 1;
+                                }
+                                self.split_basic_outputs.extend(
+                                    split_outputs
+                                        .into_iter()
+                                        .map(|output| (header.clone(), output)),
+                                );
+                                // if there was a remainder, the original output is returned for the
+                                // iterator, possibly with a modified amount; else, continue the
+                                // loop
+                                if let Some(original_output) = original_output_opt {
+                                    *inner = original_output;
+                                } else {
+                                    continue;
+                                }
+                            };
+                        }
+                    }
+                }
+            }
+            return Some(output);
+        }
+        // Now that we are out of the loop we collect the processed outputs from the
+        // timelock filter and try to fulfill the target.
+        // First, resolve timelocks SwapSplit operations, taking those out from
+        // timelocks; the ordered_timelock_candidates is ordered by timestamp
+        // and we want to take the latest ones first.
+        while let Some((header, mut inner)) = self.ordered_timelock_candidates.pop() {
+            // We know that all of them are timelocked basic outputs
+            let timelocked_basic_output = inner.as_basic();
+            let uc = timelocked_basic_output.unlock_conditions();
+            // Get destination address and mutable timelocked tokens target
+            if let Some(destinations) = self
+                .swap_split_map
+                .get_destination_maybe_mut(uc.address().unwrap().address())
+            {
+                // try to perform the SwapSplit operation for several destinations once all
+                // tokens timelocked targets are meet.
+                let (original_output_opt, split_outputs) = swap_split_operation(
+                    destinations.iter_by_tokens_timelocked_target_mut_filtered(),
```

Comment:
If the original output is timelocked then we shouldn't expect all funds to be split to timelocked targets. `iter_by_tokens_timelocked_target_mut_filtered` would swallow any such errors in the swap map.